rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/180_locale_dependent_mapping.yml

```diff
@@ -1,10 +1,10 @@
 ---
 "Test Index and Search locale dependent mappings / dates":
   - requires:
-      test_runner_features: ["allowed_warnings"]
+      test_runner_features: ["allowed_warnings_regex"]
   - do:
-      allowed_warnings:
-        - "Date format [E, d MMM yyyy HH:mm:ss Z] contains textual field specifiers that could change in JDK 23"
+      allowed_warnings_regex:
+        - "Date format \\[E, d MMM yyyy HH:mm:ss Z] contains textual field specifiers that could change in JDK 23.*"
```

Comment:
Does the closing square bracket also need to be escaped?

Reply:
Nope, `]` is only a special character if it closes a `[`. This is implied, but not explicit, in https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html#cc:
> Note that a different set of metacharacters are in effect inside a character class than outside a character class